### PR TITLE
Auto remove listener

### PR DESCRIPTION
**Pull request by Piasy**

### CODE REVIEW

File: keyboardvisibilityevent/build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ android {
     buildToolsVersion "24.0.1"
 
     defaultConfig {
-        minSdkVersion 9
+        minSdkVersion 16
```
Comment (yshrsmz) [inline keyboardvisibilityevent/build.gradle]: I didn't notice this in previous PR, but this is a HUGE change. fmm...

Comment (Piasy) [inline keyboardvisibilityevent/build.gradle]: This is required by `removeOnGlobalLayoutListener` :(

Comment (yshrsmz) [inline keyboardvisibilityevent/build.gradle]: we should use `removeGlobalOnLayoutListener` for older version. 

and `Application.ActivityLifecycleCallbacks` requires API 14 :(
...But maybe it's time to forget about pre-ICS ;)

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.java [hunk 2]
```diff
@@ -61,6 +60,15 @@ public void onGlobalLayout() {
 
                         listener.onVisibilityChanged(isOpen);
                     }
+                };
+        activityRoot.getViewTreeObserver().addOnGlobalLayoutListener(layoutListener);
+        activity.getApplication()
+                .registerActivityLifecycleCallbacks(new AutoActivityLifecycleCallback(activity) {
+                    @Override
+                    protected void onTargetActivityDestroyed() {
+                        activityRoot.getViewTreeObserver()
```
Comment (yshrsmz) [inline keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.java]: could you use `removeGlobalOnLayoutListener` for older version?

File: keyboardvisibilityevent/build.gradle (hunk 1 above)
Comment (Piasy) [inline keyboardvisibilityevent/build.gradle]: Maybe API 14 is better?

Comment (yshrsmz) [inline keyboardvisibilityevent/build.gradle]: yes please!

File: keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.java (hunk 2 above)
Comment (Piasy) [inline keyboardvisibilityevent/src/main/java/net/yslibrary/android/keyboardvisibilityevent/KeyboardVisibilityEvent.java]: No problem, will change min API to 14.

**Comment by yshrsmz** [comment]
thank you!

Since upgrading minSdk is a breaking change, this will be released as 2.0.0.

I'm planning to move to jcenter so it will take few days before publicly available.

**Pull request merged**